### DISCUSSION TITLE
Laowa Macro lens calibration

### DISCUSSION THREAD

**Discussion by sharktacos** [lensfun/lensfun#2377] [Q&A] [answered]
I'd like to upload calibration pictures for a Laowa 85mm F5.6 2:1 Ultra Macro lens. Instructions say "Test shots need to be taken at a distance of at least 5 metres."
Which, for a macro lens, this does not practically make sense. What would be the best way to provide these pics for a macro lens?

**Comment by tuxfanx** [comment, marked as answer]
A minimum distance of 8m is required for a TCA calibration ([see here](https://wilson.bronger.org/lens_calibration_tutorial/)).
It is an APO lens. Is the TCA really that strong? The distortion will also be low at this focal length.
As a first step, I would recommend the normal procedure.
The advanced version of profile creation can be used for vignetting  ([Vignetting correction for the really diligent](https://wilson.bronger.org/lens_calibration_tutorial/#id13)). 
Unfortunately, lensfun does not offer such distance-related correction for TCA and distortion.
If the profiles we have created for distortion and TCA in the macro range should not work well for you, I suggest that you create separate profiles for them yourself in a first step.

Here you will find some hints:
[calibration with lens_calibrate.py and taking pictures](https://pixls.us/articles/create-lens-calibration-data-for-lensfun/)
[calibration and taking pictures](https://wilson.bronger.org/calibration)
[How anti-distortion images should look like](https://wilson.bronger.org/calibration/target_tips)
[video 1 calibration with Hugin](https://kameratrollet.se/create-your-own-lens-correction-data-for-lensfun/)
[video 2 calibration with Hugin](https://www.youtube.com/watch?v=_p0fR-nH6cU)
[lensfuns coverage](https://wilson.bronger.org/lensfun_coverage.html)
[lensfuns database](https://github.com/lensfun/lensfun/tree/master/data/db)
[calibration with calibrate.py](https://wilson.bronger.org/lens_calibration_tutorial/)
[How anti-distortion images should look like](https://wilson.bronger.org/calibration/target_tips)
